GHermeneus/include/GHermeneus/Machine.h

```diff
@@ -5,4 +5,185 @@
+                }});
+
+            // Sort the extracted commandlines
+            std::sort(std::execution::par, cmdlines.begin(),
+                      cmdlines.end()); // Todo: check performance of diff execution policies
+        }
+
+        /*!
+         * @brief output the machines instruction vector to an output stream as in CSV format
+         * @param os Output stream
+         * @param machine A machine of type Machine<SSV_T, T>
+         * @return an output stream
+         */
+        friend std::ostream& operator<<(std::ostream& os, const Machine<SSV_T, T>& machine)
+        {
+            for (const Instruction<SSV_T, T>& cmdline : machine.cmdlines)
+            {
+                os << "line: " << cmdline.line_no << " command: " << cmdline.cmd << " Parameters ->";
+                for (const Parameter<T>& param : cmdline.params)
+                {
+                    os << " " << param.param << " = " << std::to_string(param.value);
+                }
+                os << std::endl;
+            }
+            return os;
+        };
+
+        /*!
+         * @brief Output the GCode stream to the Machines Instruction Vector
+         * @param machine The machine of type Machine<SSV_T, T>
+         * @param GCode the GCode as a string_view stream
+         * @return A machine of type Machine<SSV_T, T>
+         */
+        friend Machine<SSV_T, T>& operator<<(Machine<SSV_T, T>& machine, const std::string_view& GCode)
+        {
+            machine.parse(GCode);
+            return machine;
+        };
+
+    private:
+
+        /*!
+         * @brief extract the individual lines from the GCode
+         * @param GCode a string_view containing the GCode
+         */
+        void extractLines(const std::string_view& GCode)
+        {
+            // TODO: keep in mind CRLF and LF
+            lines = gcode
+                    | ranges::views::split('\n')
+                    | ranges::views::transform([](auto&& line) {
+                        return std::string_view(&*line.begin(), ranges::distance(line));})
+                    | ranges::views::enumerate
+                    | ranges::to_vector;
+        };
+
+        /*!
+         * @brief Extract an instruction of type Instruction<SSV_T, T> from a GCode line. If the line contains no
+         *  instruction it is a std::nullopt
+         * @param GCodeline string_view containing the line with command, the relevant parameters and/or a comment.
+         * @return an optional instruction of type Instruction<SSV_T, T>
+         */
+        [[nodiscard]] static std::optional<Instruction<SSV_T, T>> extractCmd(const Line& GCodeline)
+        {
+#ifndef NDEBUG
+            std::cout << GCodeline.first << ": " << GCodeline.second << std::endl; // Todo: Use a proper logger
+#endif
+            auto &&[lineno, gcode] = GCodeline;
+
+            // Split line in instruction and comment
+            auto split_line = gcode | ranges::views::split(';');
```

Comment:
UM has some useful, and sometimes even _machine actionable_ information in the comments. Out of scope currently (probably?), but will you handle those in the future as well?

Reply:
I know and it is indeed out of scope for now. It will be relatively simple to add in a similar way as the commands are interpreted